Sources/Commands/PackageCommands/AddSetting.swift

```diff
@@ -0,0 +1,148 @@
+            case experimentalFeature
+            case upcomingFeature
+            case languageMode
+            case strictMemorySafety
+        }
+
+        package static let configuration = CommandConfiguration(
+            abstract: "Add a new setting to the manifest"
+        )
+
+        @OptionGroup(visibility: .hidden)
+        var globalOptions: GlobalOptions
+
+        @Option(help: "The target to add the setting to")
+        var target: String
+
+        @Option(
+            name: .customLong("swift"),
+            parsing: .unconditionalSingleValue,
+            help: "The Swift language setting(s) to add. Supported settings: \(SwiftSetting.allCases.map(\.rawValue))"
+        )
+        var _swiftSettings: [String]
+
+        var swiftSettings: [(SwiftSetting, String)] {
+            get throws {
+                var settings: [(SwiftSetting, String)] = []
+                for rawSetting in self._swiftSettings {
+                    let (name, value) = rawSetting.split("=")
+
+                    guard let setting = SwiftSetting(rawValue: name) else {
+                        throw ValidationError("Unknown Swift language setting: \(name)")
+                    }
+
+                    settings.append((setting, value))
+                }
+
+                return settings
+            }
+        }
+
+        func run(_ swiftCommandState: SwiftCommandState) throws {
+            let workspace = try swiftCommandState.getActiveWorkspace()
+            guard let packagePath = try swiftCommandState.getWorkspaceRoot().packages.first else {
+                throw StringError("unknown package")
+            }
+
+            try self.applyEdits(packagePath: packagePath, workspace: workspace)
+        }
+
+        private func applyEdits(
+            packagePath: Basics.AbsolutePath,
+            workspace: Workspace
+        ) throws {
+            // Load the manifest file
+            let fileSystem = workspace.fileSystem
+            let manifestPath = packagePath.appending(component: Manifest.filename)
+
+            for (setting, value) in try self.swiftSettings {
+                let manifestContents: ByteString
+                do {
+                    manifestContents = try fileSystem.readFileContents(manifestPath)
+                } catch {
+                    throw StringError("cannot find package manifest in \(manifestPath)")
+                }
+
+                // Parse the manifest.
+                let manifestSyntax = manifestContents.withData { data in
+                    data.withUnsafeBytes { buffer in
+                        buffer.withMemoryRebound(to: UInt8.self) { buffer in
+                            Parser.parse(source: buffer)
+                        }
+                    }
+                }
+
+                let editResult: PackageEditResult
+
+                switch setting {
+                case .experimentalFeature:
+                    editResult = try AddSwiftSetting.experimentalFeature(
```

Comment:
What happens if the user adds the same setting a second time? And if that setting has a different value than the already added setting?

Reply:
The compiler would take the last value.

Reply:
Yeah, I looked into that while working on that and without some support from swift-syntax to match the nodes in general we won't be able to do match...

Reply:
I didn't do that because it would lead to false positive warnings since we cannot really figure out what is what so if they have the setting with different conditions we'll just keep warning about it for no reason.